### ISSUE TITLE
iOS requires cocoapods because of Lottie dependency. Any other option?

### ISSUE THREAD

**Issue by MyGuySi** [airbnb/lottie-react-native#19]
Hi, I'm just getting a basic example set up but I can't seem to get a visible animation on screen..

```javascript
import React from 'react';
import {
  AppRegistry,
  View,
} from 'react-native';
import Animation from 'lottie-react-native';

class LottieExample extends React.Component {
  componentDidMount() {
    this.animation.play();
  }

  render() {
    return (
      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <View style={{ borderWidth: 1, borderColor: 'black' }}>
          <Animation
            ref={animation => {this.animation = animation}}
            style={{ width: 200, height: 200, }}
            source={require('./animations/HamburgerArrow.json')}
          />
        </View>
      </View>
    );
  }
}

AppRegistry.registerComponent('LottieExample', () => LottieExample);
```

Any ideas? It's more or less a copy & paste from the README

**Comment by AlexDM0** [comment]
I had the same issue. Turns out npm install lottie-react-native does not download the ios/android libs themselves ( I only tested on iOS ). I don't use pods so I just cloned the lottie-ios, dragged the framework into my app and done. I also added the changes of pull req #20 to make sure everything compiled.

Hope that helps!

**Comment by MyGuySi** [comment]
So you dragged `LottieReactNative.xcodeproj` across to the Libraries directory, and added the Header Search Paths to the Build Settings?

I'm still getting failed builds..

```
/Users/simon/dev-test/LottieExample/node_modules/lottie-react-native/lib/ios/LottieReactNative/LottieContainerView.h:10:9: fatal error: 'RCTView.h' file not found
#import "RCTView.h"

/Users/simon/dev-test/LottieExample/node_modules/lottie-react-native/lib/ios/LottieReactNative/LottieAnimationViewManager.h:9:9: fatal error: 'RCTViewManager.h' file not found
#import "RCTViewManager.h"
```

**Comment by AlexDM0** [comment]
I did the following:

- drag the LottieReactNative.xcodeproj from node_modules/lottie-react-native/lib/ios into the Libraries folder of my app
- I clicked on it in my Libaries folder and clicked on the target of LottieReactNative, went to the Build Settings and in that header search path I added:
```
$(SRCROOT)/../../../react-native/React
$(SRCROOT)/../../../react-native/ReactCommon
``` 

Then I went to my app target linked Frameworks and Libraries and added libLottieReactNative.a from the list you get if you press +.

At this point it couldnt resolve #import <Lottie/Lottie.h>.

So I cloned https://github.com/airbnb/lottie-ios to a folder not in my app project. I then dragged the Lottie.xcodeproj into my Libraries group and added Lottie.framework to the linked Frameworks and Libraries.

At that point everything compiled and the animations were showing.

Essentially what happened was that LottieReactNative could not find react native (hence the header search paths on that proj) and it only wraps around the ios (and I'd assume android) libs. It did not contain actual code for animation. So I imported the lib it was trying to import and that fixed it.

Cheers

**Comment by MyGuySi** [comment]
Ok so I got as far as adding libLottieReatNative.a to Linked Frameworks and Libraries but I get..

```
Duplicate interface definition for class 'RCTView'
RCTView.h
```

and a bunch of 

```
Property has a previous declaration
RCTView.h
```

So is the preferred way to use Cocoapods?

**Comment by AlexDM0** [comment]
Maybe try cleaning your project?

**Comment by MyGuySi** [comment]
Yeah I did clean the project, still get those errors. Is this an issue with RN 0.40?

**Comment by AlexDM0** [comment]
I'm using 0.36.1, I see this project is using 0.38.. Could be?

**Comment by AlexDM0** [comment]
You made sure to not add these:
$(SRCROOT)/../../../react-native/React
$(SRCROOT)/../../../react-native/ReactCommon
to YOUR build settings but to the LottieReactNative.xcodeproj settings?

**Comment by MyGuySi** [comment]
Yeah, I think it's because the header paths changed in RN 0.40, I'll try updating them manually and report back

**Comment by MyGuySi** [comment]
Ok I've updated the header file paths and now I have even more errors. I think I should open a new issue for this though. I think if it weren't for that it should be working now as I no longer get any unresolved path errors, so I'll close this issue.

**Comment by lelandrichardson** [comment]
I'm going to open this back up because I think that this is still an issue (the main issue being that using Lottie as a static library right now requires cocoapods and people might not want to use that...)

**Comment by indivisable** [comment]
@AlexDM0 Thanks, Solved #25

**Comment by ABleas** [comment]
I'm also having the issue with duplicate references, on RN 0.41.1, while trying to add the lottie project manually (without Pods). Anyone figure out a solution for that? Do I need to roll back to 0.38 for this?

**Comment by lelandrichardson** [comment]
@ABleas can you walk me through what you've done so far? This should be possible, albeit a bit manual. Looking for a better solution (might include Lottie ios source in the npm published package) but haven't done this yet.

You should be able to follow the steps @indivisable referenced above in #31 and get it working w/out CocoaPods.

**Comment by lelandrichardson** [comment]
FYI, I think we've got a good solution to this. Going to try and get it fixed today.

**Comment by ABleas** [comment]
I did follow the steps in #31, except that using `npm install lottie-react-native` seemed to give me the same issues @MyGuySi was having with duplicate references. 

Not sure if npm was giving the latest GitHub version last night but instead I checked out this repo in a separate directory, added the LottieReactNative.xcodeproj to my project's Libraries, and added the dependency *libLottieReactNative.a* manually (without linking).

I guess I could have copied the project into my node_modules folder and just linked that... Anyway it works for now!
Looking forward to the fix :)
